### PR TITLE
Bug fix search table filter invalid char

### PR DESCRIPTION
**Pull request by PaschalisDim**
### Summary of Changes

_Include a summary of changes then remove this line_
When a user was passing through postman or the UI in the search_table endpoint a request with query_term= '' and filters: {schema: ["*a:a*"]}  like in the values for each of the filters if there was ':' or  '/' then it would return 500s.

### Tests

I tried to add tests in the test_search_table_filter, but it was a bit of a pain in the ass. Spent quite some time but adding a test seems to me confusing or even itself buggy over there.
My intention was to add something like:
 @patch('search_service.api.document.reqparse.RequestParser')
    @patch('search_service.api.table.get_proxy_client')
    def test_post2(self, get_proxy: MagicMock, RequestParser: MagicMock) -> None:
        RequestParser().parse_args.return_value = dict(index=self.mock_index,
                                                       page_index=self.mock_page_index,
                                                       query_term=self.mock_empty_term,
                                                       search_request=self.mock_bad_search_request)

        response = self.app.test_client().post(self.url)
        self.assertEqual(response.status_code, 400)

where in the setUp we have defined:
        self.mock_empty_term = ''
  self.mock_bad_search_request = {
            'type': 'AND',
            'filters': {
                'database': ['d:b']
            }
        }

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes `make test`

### CODE REVIEW

File: search_service/api/table.py [hunk 1]
```diff
@@ -161,6 +162,16 @@ def post(self) -> Iterable[Any]:
             msg = 'The search request payload is not available in the request'
             return {'message': msg}, HTTPStatus.BAD_REQUEST
 
+        if 'filters' in search_request:
```
Comment (feng-tao) [inline search_service/api/table.py:165]: is it an issue on ES or both ES and Atlas? If only ES, could we do it on the ES proxy?

Also we need a unit test for this check.

**Comment by ttannis** [comment]
@PaschalisDim thanks for raising this. For this case my thoughts are as follows cc @feng-tao :

1. I think regardless do what the frontend service is capable of doing (i.e. - blocking a bad request from ever being sent in the first place) that the search service should still own the logic for checking for these cases. Adhering to separation of concerns, the frontend is responsible for sending the request and updating the UI in response to that request. It should do so without having to know anything about how ElasticSearch works.

    To improve the user experience however, I'd propose that we continue with a fix in the search service *AND* eventually update the frontend such that it will show these custom error messages to the user. So someone inputs bad filter syntax, and instead of search results they will see the message from search like `The search filters contain an invalid character: '/'`. We would want to make sure to have helpful and easy to understand error messages whenever possible.

2.  As Tao hinted at, I'm cautious of applying certain checks in `search_service/api/table.py` that ends up affecting all proxies. I think this file should at most verify the request payload according to the API and then pass the information to the proxies and let the proxies implement what they need. I recognize this would mean that we'd end up doubling fixing potential bugs in atlas proxy and elastic search proxy, so perhaps a good middle ground is for the PR discussion to always include both ES and Atlas users to decide where the fix should go. If the other side does not weigh in, then we default to applying it to only one.

Let me know your thoughts! Once we're set on this I can help take a look at the unit tests

**Comment by PaschalisDim** [comment]
Hi @ttannis @feng-tao 
Thanks for jumping in to review this one.
In our case we use ES, and thus I just made the assumption that this may be affecting the rest of the implementations. It makes sense since you guys are the owners of the codebase and have better understanding to address your comments and move this functionality for the time being in the right proxy, and maybe duplicate if needed in the atlas case.
I also agree that the backend should own the checks and we should not be based on the assumption that the UI will be sending the right type of requests.
So, as an action point,
I will move this functionality in the ES proxy,
and try to get lucky with the unittests (to prove with them that this is a problem and to fix it as well)

**Comment by PaschalisDim** [comment]
@ttannis 
Kindly asking for review here.
Paschalis

**Comment by feng-tao** [comment]
thanks @PaschalisDim

**Comment by ttannis** [comment]
> @ttannis
> Kindly asking for review here.
> Paschalis

Thanks for the contribution.

**Pull request merged**